fastddsspy_participants/test/blackbox/model/ParticipantDatabaseTest.cpp

```diff
@@ -0,0 +1,402 @@
+
+    // Create DDS Pipe
+    std::shared_ptr<DdsPipe> pipe = create_pipe(model, dds_participant);
+    pipe->enable();
+
+    auto dds_reader = dds_participant->participant_reader;
+    ASSERT_NE(dds_reader, nullptr);
+
+    // Send message
+    spy::participants::ParticipantInfoData new_data;
+    random_participant_info(new_data);
+    dds_reader->simulate_data_reception(
+        std::make_unique<spy::participants::ParticipantInfoData>(new_data));
+
+    // Wait
+    std::this_thread::sleep_for(std::chrono::milliseconds(test::WAIT_MS));
+
+    // Check information
+    ASSERT_EQ(model->participant_database_.at(new_data.info.guid).guid, new_data.info.guid);
+    ASSERT_EQ(model->participant_database_.size(), 1);
+}
+
+TEST(ParticipantDatabaseTest, active_true)
+{
+    // Create Model
+    std::shared_ptr<spy::participants::SpyModel> model =
+            std::make_shared<spy::participants::SpyModel>();
+
+    const types::ParticipantId id_ = "DDS Spy mock participant";
+
+    std::shared_ptr<test::SpyDdsParticipantMock> dds_participant =
+            std::make_shared<test::SpyDdsParticipantMock>(id_);
+
+    // Create DDS Pipe
+    std::shared_ptr<DdsPipe> pipe = create_pipe(model, dds_participant);
+    pipe->enable();
+
+    auto dds_reader = dds_participant->participant_reader;
+    ASSERT_NE(dds_reader, nullptr);
+
+    // Send message
+    spy::participants::ParticipantInfoData new_data;
+    random_participant_info(new_data);
+    dds_reader->simulate_data_reception(
+        std::make_unique<spy::participants::ParticipantInfoData>(new_data));
+
+    // Wait
+    std::this_thread::sleep_for(std::chrono::milliseconds(test::WAIT_MS));
+
+    // Check information
+    ASSERT_TRUE(model->participant_database_.at(new_data.info.guid).active);
+    ASSERT_EQ(model->participant_database_.size(), 1);
+}
+
+TEST(ParticipantDatabaseTest, active_false)
+{
+    // Create Model
+    std::shared_ptr<spy::participants::SpyModel> model =
+            std::make_shared<spy::participants::SpyModel>();
+
+    const types::ParticipantId id_ = "DDS Spy mock participant";
+
+    std::shared_ptr<test::SpyDdsParticipantMock> dds_participant =
+            std::make_shared<test::SpyDdsParticipantMock>(id_);
+
+    // Create DDS Pipe
+    std::shared_ptr<DdsPipe> pipe = create_pipe(model, dds_participant);
+    pipe->enable();
+
+    auto dds_reader = dds_participant->participant_reader;
+    ASSERT_NE(dds_reader, nullptr);
+
+    // Send message
+    spy::participants::ParticipantInfoData new_data;
+    random_participant_info(new_data, false);
+    dds_reader->simulate_data_reception(
+        std::make_unique<spy::participants::ParticipantInfoData>(new_data));
+
+    // Wait
+    std::this_thread::sleep_for(std::chrono::milliseconds(test::WAIT_MS));
+
+    // Check information
+    ASSERT_FALSE(model->participant_database_.at(new_data.info.guid).active);
+
+    ASSERT_EQ(model->participant_database_.size(), 1);
+}
+
+TEST(ParticipantDatabaseTest, n_msgs_name)
```

Comment:
It would be nice to have a test case that changes, rather than adds, participant info.
For example, the most common case will be the deactivation of a Participant.